src/main/java/org/beanpod/switchboard/dao/StreamDaoImpl.java

```diff
@@ -37,7 +43,19 @@ public StreamDto saveStream(StreamDto streamDto) {
     }
 
     StreamEntity streamEntity = mapper.toEntity(streamDto);
-    return mapper.toDto(streamRepository.save(streamEntity));
+    StreamDto streamDto1 = mapper.toDto(streamRepository.save(streamEntity));
+
+    // Save an empty stream stat when saving a stream
+    if (streamDto.getStreamStat() == null) {
+      StreamStatEntity streamStatBuild =
+          StreamStatEntity.builder().stream(mapper.toEntity(streamDto1))
+              .id(streamDto1.getId())
+              .build();
+      streamStatRepository.save(streamStatBuild);
+      streamDto1.setStreamStat(statMapper.toDto(streamStatBuild));
```

Comment:
What does this do since the streamDto has already been saved without the stream stats?

Reply:
it checks when you create a stream object, are you saving a stream stat object along side with it on the create request? No? then create an empty streamstat object and save it to the DB.
Yes? then good the stream + the stat should get saved